### DISCUSSION TITLE
Usage of @Issue annotation and where clause

### DISCUSSION THREAD

**Discussion by AnthonyTonda** [spockframework/spock#1560] [General]
I would like to know if you are planning to handle this:

@Issue(“#format test”)
def “method using where clause #format test” {
   Given…
   Then…
   Where
       #format
       xml
       json
}

We will have two tests with the format xml then json
We will have also the sentence having the token #format being replace by xml then by json
Will you apply also for the annotation @issue having also the token #format ?

thanks in advance for your reply
Tonda

**Comment by Vampire** [comment]
For what would that be useful?
You could probably even write an own global Spockextension that does it, all the `@Issue` extension does is setting Tags, besides serving as documentation itself which could be used by some reporting extension or similar.
But the intention of the `@Issue` annotation is to write there the issue that is tested with the test, like the link to the GitHub issue, or alternatively just the issue number.
I don't see how there the data variables would be any helpful in 99% of the cases.

**Reply by AnthonyTonda** [reply]
Thanks for the answer.
If by that design, 3 tests are running and by your explanation, you link to only one test
Shouldn’t we link by 3 three distinct links ?

as you said I should be able to cover by SpockExtension, please share the tutorial/the link how to write my own global spockextension.

**Reply by AnthonyTonda** [reply]
Me I would like link a test to a given link which it would be in the where clause

**Reply by Vampire** [reply]
> Shouldn’t we link by 3 three distinct links ?

Why?
You could also write 10 different test methods to cover one issue and all can have the same value for the `@Issue` annotation.
There is absolutely no reason to need to have distinct values.

> please share the tutorial/the link how to write my own global spockextension.

https://spockframework.org/spock/docs/2.3/all_in_one.html#_writing_custom_extensions
But actually while thinking about it, there is probably not much useful you could do.

> Me I would like link a test to a given link which it would be in the where clause

The main question probably is, what you would want to do with that information.
The built-in extension just sets Spock tags derived from the `@Issue` values.

**Reply by AnthonyTonda** [reply]
Let’s agree that we have different point of view of using the annotation @issue 
For a Story we have multiple Issues or Tests
A test can be parameterized. I want to be link to distinct issue/test

**Reply by AnthonyTonda** [reply]
"The built-in extension just sets Spock tags derived from the @Issue values."

is it possible to set the @Issue value from the where block ?

**Reply by Vampire** [reply]
No

**Reply by AnthonyTonda** [reply]
Can you explain, please. Why it is no ?

**Reply by Vampire** [reply]
Because it is an annotation.
Annotation values need to be compile-time constants.
